client/client.go

```diff
@@ -264,8 +264,8 @@ type Client struct {
 	// Service Identity tokens through Nomad Server.
 	tokensClient consulApi.ServiceIdentityAPI
 
-	// vaultClient is used to interact with Vault for token and secret renewals
-	vaultClient vaultclient.VaultClient
+	// vaultClients is used to interact with Vault for token and secret renewals
+	vaultClients map[string]vaultclient.VaultClient
```

Comment:
I think it's fine, but just noting it here in case I'm mistaken. We don't need a lock for this map because the only mutation happens in `NewClient` and accessing the map requires `NewClient` to be called beforehand.

Reply:
Right. `NewClient` is only ever called in `setupVaultClients`